### PR TITLE
Dict bool keys

### PR DESCRIPTION
**Pull request by kabra1110**
Work in progress. Towards #1710.

### CODE REVIEW

File: src/libasr/codegen/llvm_utils.cpp [hunk 1]
```diff
@@ -1962,6 +1962,21 @@ namespace LCompilers {
...
+                
+                llvm_utils->create_if_else(is_key_one, [&]() {
+                    LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 1)), hash_value);
+                }, [=]() {
+                    LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), hash_value);
+                });
+                return hash_value;
+            }
```
Comment (czgdp1807) [inline src/libasr/codegen/llvm_utils.cpp:1979]: This logic is correct but the if-else check seems unnecessary. A simple cast from `i1` to `i32` should have done the job. We will see in our tomorrow's meeting how can we do that.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
Add tests like looping and writing/reading again and again from a bool dictionary. Value can be of different types like, integer, float, string, etc.

File: integration_tests/test_dict_bool.py
Comment (czgdp1807) [inline integration_tests/test_dict_bool.py:1]: We are doing `d[True]` and `d[False]`. Let's **also** do something like,

```
d[i%2 == 0] += 1
```

And check the count of odd and even numbers for some range `[0, n)`. To make sure things work correctly for runtime boolean values.

File: tests/reference/asr-test_dict_bool-6f905fd.stdout
Comment (czgdp1807) [inline tests/reference/asr-test_dict_bool-6f905fd.stdout:1]: Did you generate these by hand?

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: tests/reference/asr-test_dict_bool-6f905fd.stdout
Comment (kabra1110) [reply tests/reference/asr-test_dict_bool-6f905fd.stdout:1]: No, I added (and later removed) a test in `./tests/tests.toml` with `asr` and `asr_json` options.

Comment (czgdp1807) [reply tests/reference/asr-test_dict_bool-6f905fd.stdout:1]: Ah I see.Then you can remove these `.stdout` files.

File: integration_tests/test_dict_bool.py
Comment (kabra1110) [reply integration_tests/test_dict_bool.py:1]: Looks like `+=` isn't currently supported for `dict` yet. I have added this test with `=`.

Comment (czgdp1807) [reply integration_tests/test_dict_bool.py:1]: No worries.

> Looks like `+=` isn't currently supported for `dict` yet.

Let's support this in a separate PR, once you complete this and your other open PR.

File: tests/reference/asr_json-test_dict_bool-8307f87.stdout
Comment (czgdp1807) [inline tests/reference/asr_json-test_dict_bool-8307f87.stdout:1]: Delete this as well.

File: tests/reference/asr_json-test_dict_bool-8307f87.json
Comment (czgdp1807) [inline tests/reference/asr_json-test_dict_bool-8307f87.json:1]: This one too.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
If `llvm` test is enabled in `integration_tests/CMakeLists.txt` then most of the time registering the same test in `.toml` file is not needed.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**